### PR TITLE
Synchronous OCTEON Sniffer Support

### PR DESCRIPTION
**Pull request by ejohnstown**
Added the synchronous use of the OCTEON to the sniffer.

### CODE REVIEW

File: wolfssl/wolfcrypt/port/cavium/cavium_octeon_sync.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/* cavium_octeon.h
+ *
+ * Copyright (C) 2006-2019 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL. (formerly known as CyaSSL)
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
```
Comment (dgarske) [inline wolfssl/wolfcrypt/port/cavium/cavium_octeon_sync.h]: Please change to version 2 license.

File: wolfssl/wolfcrypt/aes.h [hunk 2]
```diff
@@ -198,6 +201,7 @@ struct Aes {
 #if defined(WOLF_CRYPTO_CB) || (defined(WOLFSSL_DEVCRYPTO) && \
     (defined(WOLFSSL_DEVCRYPTO_AES) || defined(WOLFSSL_DEVCRYPTO_CBC)))
     word32 devKey[AES_MAX_KEY_SIZE/WOLFSSL_BIT_SIZE/sizeof(word32)]; /* raw key */
+    int    keySet;
```
Comment (dgarske) [inline wolfssl/wolfcrypt/aes.h]: The `keySet` is Octeon specific. Can we add a `HAVE_CAVIUM_OCTEON_SYNC` check around this?

File: wolfcrypt/src/port/cavium/cavium_octeon_sync.c [hunk 3]
```diff
@@ -0,0 +1,995 @@
+/* cavium_octeon.c
+ *
+ * Copyright(C) 2006-2019 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.(formerly known as CyaSSL)
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
```
Comment (dgarske) [inline wolfcrypt/src/port/cavium/cavium_octeon_sync.c]: Check License version.

File: wolfcrypt/src/port/cavium/README_Octeon.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
+# Octeon III
```
Comment (dgarske) [inline wolfcrypt/src/port/cavium/README_Octeon.md]: I'm not sure if we can/should make these instructions public. Also please replace the "dgarske" references with `"<username>"`.

File: wolfssl/wolfcrypt/port/cavium/cavium_octeon_sync.h (hunk 1 above)
Comment (ejohnstown) [reply wolfssl/wolfcrypt/port/cavium/cavium_octeon_sync.h]: Fixed, and updated the filename at the top of the file.

File: wolfcrypt/src/port/cavium/cavium_octeon_sync.c (hunk 3 above)
Comment (ejohnstown) [reply wolfcrypt/src/port/cavium/cavium_octeon_sync.c]: Fixed, and updated the filename at the top of the file.

File: wolfssl/wolfcrypt/aes.h (hunk 2 above)
Comment (ejohnstown) [reply wolfssl/wolfcrypt/aes.h]: Yes.

File: wolfcrypt/src/port/cavium/README_Octeon.md (hunk 4 above)
Comment (ejohnstown) [reply wolfcrypt/src/port/cavium/README_Octeon.md]: I replaced this file with an Octeon version of the other README.

**Comment by kaleb-himes** [comment]
NOTE: The wolfSSL website ca signer was updated, Jacob will open a PR soon with the new CA in wolfssl-website-ca.pem

File: wolfssl/wolfcrypt/port/intel/quickassist_sync.h [hunk 5]
```diff
@@ -155,16 +155,16 @@ typedef struct IntelQaDev {
...
-WOLFSSL_LOCAL void IntelQaHardwareStop(void);
-WOLFSSL_LOCAL int IntelQaInit(void*);
-WOLFSSL_LOCAL void IntelQaDeInit(int);
-WOLFSSL_LOCAL int IntelQaNumInstances(void);
-WOLFSSL_LOCAL int IntelQaOpen(IntelQaDev*, int);
-WOLFSSL_LOCAL void IntelQaClose(IntelQaDev*);
-WOLFSSL_LOCAL int IntelQaDevCopy(IntelQaDev*, IntelQaDev*);
-WOLFSSL_LOCAL int IntelQaPoll(IntelQaDev*);
-WOLFSSL_LOCAL int IntelQaGetCyInstanceCount(void);
+WOLFSSL_API int IntelQaHardwareStart(const char*, int);
```
Comment (JacobBarthelmeh) [inline wolfssl/wolfcrypt/port/intel/quickassist_sync.h]: Can these be made to be wolfSSL_* when switching over to public facing?

**Pull request merged**